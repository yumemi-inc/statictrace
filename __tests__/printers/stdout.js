const { run } = require('../../build/lib');
const dotenv = require('dotenv');
const path = require('path');

dotenv.config();

const configPath = path.join(
  process.env.TS_PROJECT_TEST,
  'basic',
  'tsconfig.json'
);

it('Prints to stdout correctly', async () => {
  const output = await run(configPath);
  expect(output).toMatchSnapshot();
});
