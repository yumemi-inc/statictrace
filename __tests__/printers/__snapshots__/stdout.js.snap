// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Accepts custom printers 1`] = `
"{
  \\"Database_constructor\\": 0,
  \\"get\\": 1,
  \\"saveUser\\": 1,
  \\"post\\": 2,
  \\"startRegistration\\": 0,
  \\"processRegistration\\": 1,
  \\"someRegistrationProcedure\\": 2,
  \\"callInsideTracedFn\\": 2,
  \\"finishRegistration\\": 1,
  \\"cleanupSomething\\": 1,
  \\"begin\\": 0,
  \\"funcA\\": 0,
  \\"funcC\\": 1,
  \\"beingNestedEntrypoint\\": 0,
  \\"funcB\\": 1,
  \\"infiniteSelfRecursion\\": 1,
  \\"indirectInfiniteSelfRecursion\\": 2,
  \\"makeIndirectRecursiveCall\\": 1
}"
`;

exports[`Prints to stdout correctly 1`] = `
"=======================
Entrypoint: Database
Database_constructor
	get
	saveUser
		post

=======================
Entrypoint: Registration
startRegistration
	processRegistration
		someRegistrationProcedure
			callInsideTracedFn
	finishRegistration
		callInsideTracedFn
	cleanupSomething

=======================
Entrypoint: OuterEntrypoint
begin
	funcA
		funcC
	beingNestedEntrypoint
		funcA
			funcC
		funcB
	funcB

=======================
Entrypoint: NestedEntrypoint
beingNestedEntrypoint
	funcA
		funcC
	funcB

=======================
Entrypoint: AnotherNestedEntrypoint
funcA
	funcC

=======================
Entrypoint: InfiniteSelfRecursion
infiniteSelfRecursion
	infiniteSelfRecursion

=======================
Entrypoint: IndirectInfiniteSelfRecursion
indirectInfiniteSelfRecursion
	makeIndirectRecursiveCall
		indirectInfiniteSelfRecursion

"
`;
